### PR TITLE
Fix results star rating display not being centered when no mods are present

### LINKED ISSUE #8599: Star Rating is not centered in result screen when there are no mods
**Screenshots or videos showing encountered issue:** 
container
star rating

(Probably low priority)

### PR DESCRIPTION
**Pull request by Joehuu**
Fixes https://github.com/ppy/osu/issues/8599.

Actually this is also misaligned with mods (fix in first commit), but made title represent issue.

### CODE REVIEW

File: osu.Game.Tests/Visual/UserInterface/TestSceneModSelectOverlay.cs [hunk 1]
```diff
@@ -70,7 +70,7 @@ private void load(RulesetStore rulesets)
                     Anchor = Anchor.TopRight,
                     Origin = Anchor.TopRight,
                     AutoSizeAxes = Axes.Both,
-                    Position = new Vector2(0, 25),
+                    Position = new Vector2(-5, 25),
```
Comment (mcendu) [inline osu.Game.Tests/Visual/UserInterface/TestSceneModSelectOverlay.cs:73]: Why are these changed?

**Comment by Joehuu** [comment]
Weird, how did that happen?

Anyway, going to make a more logical change by applying margin to each mod display usage.

**Comment by peppy** [comment]
I'm confused. What exactly happened here?
